File: analyzer/src/io/github/colemakmods/keyboard/ExperienceCurve.java
package io.github.colemakmods.keyboard;

public class ExperienceCurve {

    private double pParam;

    public ExperienceCurve(double bParam) {
        this.pParam = Math.log(bParam)/Math.log(2);
        System.out.println();
        System.out.println(String.format("ExperienceCurve b=%f, p=%f", bParam, pParam));
    }

    public double getInstantCost(double freq) {
        return Math.pow(freq, pParam);
    }

    public double getIntegratedCost(double freq) {
        return Math.pow(freq, pParam+1)/(pParam+1);
    }

    public static void main(String[] args) {
        double b = 0.8;
        ExperienceCurve experienceCurve = new ExperienceCurve(b);
        for (double f = 0; f <=15; ++f) {
            System.out.println(String.format("%f :  %f  %f", f, experienceCurve.getInstantCost(f), experienceCurve.getIntegratedCost(f)));
        }
    }

}

File: analyzer/src/io/github/colemakmods/keyboard/ExperienceCurve.kt
package io.github.colemakmods.keyboard

import kotlin.math.ln
import kotlin.math.pow

class ExperienceCurve(bParam: Double) {
    private val pParam: Double

    init {
        pParam = ln(bParam) / ln(2.0)
        println()
        println(String.format("ExperienceCurve b=%f, p=%f", bParam, pParam))
    }

    fun getInstantCost(freq: Double): Double {
        return freq.pow(pParam)
    }

    fun getIntegratedCost(freq: Double): Double {
        return freq.pow(pParam + 1) / (pParam + 1)
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val b = 0.8
            val experienceCurve = ExperienceCurve(b)
            for (f in 0..15) {
                val instantCost = experienceCurve.getInstantCost(f.toDouble())
                val integratedCost = experienceCurve.getIntegratedCost(f.toDouble())
                println("$f :  $instantCost  $integratedCost")
            }
        }
    }
}
